rest-services/server/src/test/java/com/_4point/aem/docservices/rest_services/server/assembler/AssembleDocumentsTest.java

```diff
@@ -96,15 +111,17 @@ void testDoPost_HappyPath_JustForm() throws ServletException, IOException, NoSuc
 		GenerateAssemblerResultArgs generateAssemblerResultArgs = assemblePdfMock.getGenerateAssemblerResultArgs();
 		assertNotNull(generateAssemblerResultArgs.getDdx());
         AssemblerOptionsSpec assemblerOptionsSpec =
-		  generateAssemblerResultArgs.getAssemblerOptionsSpec(); assertAll(
+		  generateAssemblerResultArgs.getAssemblerOptionsSpec(); 
+        assertAll(
 		  ()->assertNotNull(assemblerOptionsSpec.getLogLevel()),
-		  ()->assertNull(assemblerOptionsSpec.isFailOnError()),
-		  ()->assertNull(assemblerOptionsSpec.isTakeOwnership()),
-		  ()->assertNull(assemblerOptionsSpec.isValidateOnly()),
-		  ()->assertNull(assemblerOptionsSpec.getDefaultStyle()),
+		  ()->assertNotNull(assemblerOptionsSpec.isFailOnError()),
+		  ()->assertNotNull(assemblerOptionsSpec.isTakeOwnership()),
+		  ()->assertNotNull(assemblerOptionsSpec.isValidateOnly()),
+		  ()->assertNotNull(assemblerOptionsSpec.getDefaultStyle()),
 		  ()->assertNotNull(assemblerOptionsSpec.getFirstBatesNumber()));	 
```

Comment:
Can we do more than `assertNotNull()` here?  Can we check that the value is actually the value that was set?  While the current code will catch some kinds of errors, just checking for null will miss if we are setting _the wrong_ value by accident (for example, if set `isValidateOnly` based on `isFailOnError` because of a copy/paste error.  By not only checking whether or not they are null but also checking that the value is correct we are going to catch more and different types of defects.

Also, it's also worth adding an additional test that tests the case where we _do not_ specify any values and make sure that the assemblerOptionSpec values are null.  That way we have tested both possible code paths.

Reply:
So, I'm still seeing a lot of red blocks in `AssembleDocuments.java` when I right-click on `AssembleDocumentsTest.java` and select Coverage As>JUnit Test.  This shows you what test cases you are missing (by showing you which lines of code are not being executed by the unit tests).

I would expect that the unit tests would cover all lines of code that are not part of exception handling.  You've done a good job at including error conditions in your test so that's very positive.  We just need to go a little farther and make sure that all the non-error code paths are also being exercised as part of the unit tests.

Reply:
added test cases for the code which was not covered earlier  in the test case.